### DISCUSSION TITLE
httpClientFactory & Session port

### DISCUSSION THREAD

**Discussion by des1sgm** [FantasticFiasco/serilog-sinks-http#282] [Ideas]
Hello, I have successfully tested the http sink with an custom httpClient. Thanks for the good work :-) 
I only noticed two things (or did I understand this wrong?)
-In the custom httpClient, the httpClient itself is initialised and not by the middleware httpclientfactory.
-And a new session port is created for each request.

Is there a solution for this?

Many thanks for your help!

**Comment by FantasticFiasco** [comment]
> In the custom httpClient, the httpClient itself is initialised and not by the middleware httpclientfactory.

Yes. The [basic authenticated HTTP client](https://github.com/FantasticFiasco/serilog-sinks-http/wiki/Custom-HTTP-client#example-the-basic-authenticated-http-client) is a pretty good example where the `HttpClient` is created in the implementation of `IHttpClient`.

> And a new session port is created for each request.

Are you referring to the problem where each time a new instance of a `HttpClient` is created, a new port is mapped? In that case we have no problem here since the implementation of `IHttpClient` only is created once, thus the same instance of `HttpClient` is reused during the lifetime of the Serilog sink.
